torchgeo/models/weights.py

```diff
@@ -0,0 +1,154 @@
+"""Pretrained weights for models."""
+import os
+from typing import Any, Dict
+
+import torch
+import torch.nn as nn
+from torchvision.models._api import Weights, WeightsEnum
+
+from torchgeo.datasets.utils import download_url
+
+__all__ = ["ResNet50_Weights"]
+
+
+class ResNet50_Weights(WeightsEnum):
+    """ResNet50 weights.
+
+    For `timm
+    <https://github.com/rwightman/pytorch-image-models>`_ implementation.
+    """
+
+    IMAGENET_RGB = Weights(
+        url=(
+            "https://github.com/rwightman/pytorch-image-models/releases/"
+            "download/v0.1-rsb-weights/resnet50_a1_0-14fe96d1.pth"
+        ),
+        transforms=nn.Identity(),
+        meta={
+            "dataset": "imagenet",
+            "ssl_method": None,
+            "repo": (
+                "https://github.com/rwightman/pytorch-image-models/"
+                "blob/main/timm/models/resnet.py"
+            ),
+        },
+    )
+    SENTINEL2_RGB_SECO = Weights(
+        url="https://zenodo.org/record/4728033/files/seco_resnet50_1m.ckpt?download=1",
+        transforms=nn.Identity(),
+        meta={
+            "ssl_method": "seco",
+            "dataset": "seco",
+            "publication": "https://arxiv.org/abs/2103.16607",
+            "repo": "https://github.com/ServiceNow/seasonal-contrast",
+        },
+    )
+
+    AERIAL_MILLIONAID_RGB = Weights(
+        url="https://drive.google.com/file/d/1K3P4_fDfcBRGqpKoSdSa6OXS4xC1xLC9",
+        transforms=nn.Identity(),
+        meta={
+            "dataset": "millionaid",
+            "ssl_method": None,
+            "repo": (
+                "https://github.com/ViTAE-Transformer/"
+                "ViTAE-Transformer-Remote-Sensing"
+            ),
+        },
+    )
+
+    SENTINEL2_ALL_MOCO = Weights(
+        url=(
+            "https://drive.google.com/file/d/1OrtPfG2wkO05bimstQ_T9Dza8z3zp8i-/"
+            "view?usp=sharing/B13_rn50_moco_0099_ckpt.pth"
+        ),
+        transforms=nn.Identity(),
+        meta={
+            "ssl_method": "moco",
+            "publication": "https://arxiv.org/abs/2211.07044",
+            "dataset": "SSL4EO-S12",
+            "repo": "https://github.com/zhu-xlab/SSL4EO-S12",
+        },
+    )
+    SENTINEL2_ALL_DINO = Weights(
+        url="https://drive.google.com/file/d/1iSHHp_cudPjZlshqWXVZj5TK74P32a2q",
+        transforms=nn.Identity(),
+        meta={
+            "ssl_method": "dino",
+            "publication": "https://arxiv.org/abs/2211.07044",
+            "dataset": "SSL4EO-S12",
+            "repo": "https://github.com/zhu-xlab/SSL4EO-S12",
+        },
+    )
+    DEFAULT = SENTINEL2_ALL_MOCO
```

Comment:
I don't think it makes sense for there to be a DEFAULT set of weights since it depends on the number of input channels.